### DISCUSSION TITLE
[Bug]: Intermittent OpenRouter Error in LibreChat (200 OK Error)

### DISCUSSION THREAD

**Discussion by rehypothecation** [danny-avila/LibreChat#4357] [Troubleshooting]
### What happened?

# Intermittent OpenRouter Error in LibreChat

## Description
When using OpenRouter in LibreChat, I'm encountering an intermittent error. The error message only shown in the webapp (not the logs) is:

```
Unexpected response from server; Status: 200 OK
```

This error occurs inconsistently, with no apparent pattern. Sometimes the application works fine, but other time  it throws this error.

## Environment
- LibreChat version: 0.7.5-rc2
- Platform: desktop-linux
- Node.js version: v22.6.0

## Behavior
Intermittently, the chat throws an "Unexpected response from server" error, despite receiving a 200 OK status.

## Additional Information
- The error seems to only occur when using OpenRouter.
- There are no visible errors in the logs.
- The issue is inconsistent and doesn't follow a clear pattern.

## Possible Causes
1. Timing issues in processing the response. There is a little pause before a response starts filling. Then it fills in a little and opens a second response "2/2"
2. Incorrect handling of 200 OK responses in certain scenarios

Any assistance would be greatly appreciated.

### Steps to Reproduce

Set up LibreChat with OpenRouter integration
Attempt to use the chat functionality
Observe that sometimes it works, while other times it produces the error message

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
2024-10-08T17:00:03.221Z debug: [OpenAIClient] chatCompletion: dropped params
{
  // 1 dropParam(s)
  dropParams: ["stop"],
    modelOptions.model: "perplexity/llama-3.1-sonar-large-128k-online",
    modelOptions.user: "[...]",
    modelOptions.stream: true,
    // 5 message(s)
    modelOptions.messages: [/* messages omitted for brevity */],
}
2024-10-08T17:00:03.222Z debug: Making request to https://openrouter.ai/api/v1/chat/completions
2024-10-08T17:00:03.225Z debug: [saveConvo] api/app/clients/BaseClient.js - saveMessageToDatabase #saveConvo
2024-10-08T17:00:09.949Z debug: [OpenAIClient] chatCompletion response
{
  provider: "Perplexity",
  object: "chat.completion",
    usage.prompt_tokens: 1478,
    usage.completion_tokens: 280,
    usage.total_tokens: 1758,
  id: "gen-[...]",
  // 1 choice(s)
  choices: [/* choices omitted for brevity */],
  created: 1728406803,
  model: "perplexity/llama-3.1-sonar-large-128k-online",
}
2024-10-08T17:00:09.949Z debug: [spendTokens] conversationId: [...] | Context: message | Token usage:
{
  promptTokens: 1574,
  completionTokens: 280,
}
2024-10-08T17:00:09.954Z debug: [spendTokens] No transactions incurred against balance
2024-10-08T17:00:09.958Z debug: [saveConvo] api/app/clients/BaseClient.js - saveMessageToDatabase #saveConvo
2024-10-08T17:00:09.971Z debug: [AskController] Request closed
```

### Screenshots

<img width="745" alt="Screenshot 2024-10-08 at 1 10 38 PM" src="https://github.com/user-attachments/assets/515274c8-03c1-4a12-8ecb-6c42349feaea">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It’s an OpenRouter issue, please let them know on their discord
